hazelcast-client/src/test/java/com/hazelcast/client/topic/ClientReliableTopicOverloadTest.java

```diff
@@ -0,0 +1,98 @@
+ */
+
+package com.hazelcast.client.topic;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.config.ClientReliableTopicConfig;
+import com.hazelcast.client.impl.HazelcastClientProxy;
+import com.hazelcast.client.proxy.ClientReliableTopicProxy;
+import com.hazelcast.client.test.TestHazelcastFactory;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.RingbufferConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.ITopic;
+import com.hazelcast.internal.serialization.SerializationService;
+import com.hazelcast.ringbuffer.Ringbuffer;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import com.hazelcast.topic.TopicOverloadPolicy;
+import com.hazelcast.topic.impl.reliable.ReliableTopicMessage;
+import com.hazelcast.topic.impl.reliable.TopicOverloadAbstractTest;
+import org.junit.After;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static com.hazelcast.test.AbstractHazelcastClassRunner.getTestMethodName;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class ClientReliableTopicOverloadTest extends TopicOverloadAbstractTest {
+
+    private TestHazelcastFactory hazelcastFactory = new TestHazelcastFactory();
+    private ClientReliableTopicProxy<String> topic;
+    private Ringbuffer<ReliableTopicMessage> ringbuffer;
+    private SerializationService serializationService;
+
+    @Override
+    public void setupCluster() {
+        Config config = new Config();
+        config.addRingBufferConfig(new RingbufferConfig("when*")
+                .setCapacity(100).setTimeToLiveSeconds(5));
+        hazelcastFactory.newHazelcastInstance(config);
+
+        ClientConfig clientConfig = new ClientConfig();
+        clientConfig.addReliableTopicConfig(new ClientReliableTopicConfig("whenError_*")
+                .setTopicOverloadPolicy(TopicOverloadPolicy.ERROR));
+        clientConfig.addReliableTopicConfig(new ClientReliableTopicConfig("whenDiscardOldest_*")
+                .setTopicOverloadPolicy(TopicOverloadPolicy.DISCARD_OLDEST));
+        clientConfig.addReliableTopicConfig(new ClientReliableTopicConfig("whenDiscardNewest_*")
+                .setTopicOverloadPolicy(TopicOverloadPolicy.DISCARD_NEWEST));
+        clientConfig.addReliableTopicConfig(new ClientReliableTopicConfig("whenBlock_*")
+                .setTopicOverloadPolicy(TopicOverloadPolicy.BLOCK));
+        HazelcastInstance client = hazelcastFactory.newHazelcastClient(clientConfig);
+
+        serializationService = ((HazelcastClientProxy) client).getSerializationService();
+
+        String topicName = getTestMethodName();
+        topic = (ClientReliableTopicProxy<String>) client.<String>getReliableTopic(topicName);
+
+        ringbuffer = topic.getRingbuffer();
+    }
+
+    @Override
```

Comment:
Isn't it easier if the TopicOverloadAbstract test has fields like ringbuffer and the subclass is responsible for initializing them? Adding these 'getters' is only code noise.